### ISSUE TITLE
ArrayIndexOutOfBoundsException crash from org.json

### ISSUE THREAD

**Issue by benjamin-bader** [amplitude/Amplitude-Android#64]
We've hit this a few times in production, never on our own devices.

I haven't looked too closely, but this seems like a bug in org.json - that index is absurd.  The call-site for `logEvent` is literally just `Amplitude.logEvent(name, obj)` where `obj` is a `JSONObject`.

Probably nothing to do but catch and log (maybe propagate in some other fashion).  My personal opinion is that this shouldn't be thrown in its current form.

```
java.lang.ArrayIndexOutOfBoundsException: length=3; index=319421760
       at java.util.ArrayList.add(ArrayList.java:124)
       at java.util.AbstractCollection.toArrayList(AbstractCollection.java:350)
       at java.util.AbstractCollection.toArray(AbstractCollection.java:339)
       at java.util.ArrayList.(ArrayList.java:97)
       at org.json.JSONObject.names(JSONObject.java:677)
       at com.amplitude.api.AmplitudeClient.cloneJSONObject(AmplitudeClient.java:885)
       at com.amplitude.api.AmplitudeClient.logEventAsync(AmplitudeClient.java:281)
       at com.amplitude.api.AmplitudeClient.logEvent(AmplitudeClient.java:250)
       at com.amplitude.api.AmplitudeClient.logEvent(AmplitudeClient.java:245)
       at com.amplitude.api.Amplitude.logEvent(Amplitude.java:60)
```

**Comment by djih** [comment]
Thanks for bringing this to our attention. We will put the fix in our next release.

**Comment by djih** [comment]
https://github.com/amplitude/Amplitude-Android/commit/f5baf6a9120337b79e56fe221f164f35d7c06b4a will be pushed in the next release

**Comment by benjamin-bader** [comment]
Thanks!

**Comment by djih** [comment]
This is released with [v2.2.0](https://github.com/amplitude/Amplitude-Android/releases/tag/v2.2.0), might take a few hours to appear on Maven Central
